src/tests/unit/common/rule-based-view-model-provider.test.ts

```diff
@@ -34,24 +34,24 @@ describe('RuleBasedViewModelProvider', () => {
     });
 
     const testStubCombinations = [
-        { isExpanded: true, isSelected: true },
-        { isExpanded: false, isSelected: false },
-        { isExpanded: true, isSelected: false },
+        { isExpanded: true, isSelected: true, isHighlighted: true },
```

Comment:
this make me think about our model:
- is it possible to have something selected but not highlighted? 
- is it possible to have something highlighted but not selected?

If it's possible, we're missing test cases here. If its **not** possible, then the follow up question is, do we need both flags?

Reply:
It is possible; good call. Will add a stub for this.